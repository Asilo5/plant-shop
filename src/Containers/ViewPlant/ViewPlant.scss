.view-plant {
    background: white;
    padding: 40px;
    width: 80%;
    margin: 45px auto;
    height: 70vh;
    overflow: scroll;
    box-shadow: 5px 5px 5px #888888;
    .buttons-section {
        display: flex;
        justify-content: space-between;
        .back-button {
          text-decoration: none;
          font-size: 20px;
          color: black;
          &:hover {
              font-weight: bold;
          }
        }
        .unloved-plant {
            border: none;
            font-size: 2em;
            color: #808080c2;
            padding: 10px 10px 0 10px;
            border: 2px solid transparent;
            &:hover {
                border: 2px solid cadetblue;
                border-radius: 50%;
                cursor: pointer;
                color: cadetblue;
            }
        }
        .loved-plant {
            @extend .unloved-plant;
            color: cadetblue;
            border: 2px solid cadetblue;
            border-radius: 50%;
            &:hover {
                color: #808080c2;
                border: 2px solid transparent;
            }
        }

    }
    .all-plant {
        display: flex;
        justify-content: space-around;
        .view-plant-info {
          h2 {
              font-size: 4em;
          }
          h3 {
              font-size: 2em;
          }
          p {
              font-size: 22px;
          }
          .delete-btn {
            text-decoration: none;
            font-size: 20px;
            padding: 5px;
            border-radius: 4px;
            border: 1px solid red;
            color: red;
            font-weight: bold;
            position: relative;
            top: 10em;
            &:hover {
                cursor: pointer;
                color: white;
                background-color: red;
            }
          }
        }
    }
}