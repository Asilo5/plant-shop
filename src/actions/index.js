export const addPlants = (setPlants) => ({
    type: 'ADD_PLANTS',
    setPlants
});

export const hasError = (error) => ({
  type: 'HAS_ERROR',
  error
});

export const newPlant = (plant) => ({
  type: 'NEW_PLANT',
  plant
})