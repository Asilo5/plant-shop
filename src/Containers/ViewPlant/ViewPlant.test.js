import React from 'react';
import { shallow } from 'enzyme';
import {ViewPlant, mapStateToProps, mapDispatchToProps} from './ViewPlant';
import { addFavouritePlant, deleteFavouritePlant } from '../../actions';

describe('ViewPlant Container', () => {

    describe('ViewPlant', () => {
        const favPlants= [1,2,3];

        it('should match snapshot', () => {
            let wrapper = shallow(<ViewPlant _id={1} 
                                             name='Aloe'
                                             price={15}
                                             image='image link'
                                             description='green queen'
                                             care='and sunshine' 
                                             favouritePlants={favPlants} 
                                             addFavouritePlant={jest.fn()} 
                                             deleteFavouritePlant={jest.fn()} />) 
    
           expect(wrapper).toMatchSnapshot();
        })
    })

    describe('mapStateToProps', () => {
        it('should return an object with current state used in component', () => {
            const mockState = {
                favouritePlants: [1,2,4]
            };

            const expected = {
                favouritePlants: [1,2,4]
            };

            const mappedProps = mapStateToProps(mockState);
            expect(mappedProps).toEqual(expected);
        })
    })

    describe('mapDispatchToProps', () => {

        const mockAddFavouritePlant = {
            name: "Aloe Queen",
            description: "Healing plant",
            care: "Lots of Love",
            price: 20,
            image: "https://target.scene7.com/is/image/Target/GUEST_be7e6313-15ac-4e1c-94b8-8af5595d6cb6?wid=488&hei=488&fmt=pjpeg",
            
        };
        const mockDeleteFavouritePlant = '1';

        const mockDispatch = jest.fn();

        it('should dispatch addFavouritePlant', () => {
            const actionToDispatch = addFavouritePlant(mockAddFavouritePlant);
            const mappedProps = mapDispatchToProps(mockDispatch);

            mappedProps.addFavouritePlant(mockAddFavouritePlant);
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
        })

        it('should dispatch deleteFavouritePlant', () => {
            const actionToDispatch = deleteFavouritePlant(mockDeleteFavouritePlant);
            const mappedProps = mapDispatchToProps(mockDispatch);

            mappedProps.deleteFavouritePlant(mockDeleteFavouritePlant);
            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
        })
    })

})
