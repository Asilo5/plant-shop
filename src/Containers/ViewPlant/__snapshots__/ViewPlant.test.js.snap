// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPlant should match snapshot 1`] = `
<section
  className="view-plant"
>
  <div
    className="buttons-section"
  >
    <Link
      className="back-button"
      to="/allPlants"
    >
      Back
    </Link>
    <button>
      <FaSeedling />
    </button>
  </div>
  <section
    className="all-plant"
  >
    <img
      alt="Aloe"
      src="image link"
    />
    <div
      className="view-plant-info"
    >
      <h2>
        Aloe
      </h2>
      <h3>
        Â£
        15
      </h3>
      <p>
        Description: 
        green queen
      </p>
      <p>
        Care: 
        and sunshine
      </p>
    </div>
  </section>
</section>
`;
